test/functional/logos_feature_bip341_sighash.py

```diff
@@ -0,0 +1,386 @@
+        tx_fan_out.vin.append(CTxIn(COutPoint(coin, 0), CScript([b'\x51'])))
+        tx_fan_out.vout = spendable_outputs
+        tx_fan_out.rehash()
+
+        node.p2p.send_txs_and_test([tx_fan_out], node)
+
+        utxo_idx = 0
+        key_idx = 0
+        for test_case in TESTCASES:
+            if test_case == 'ENABLE_REPLAY_PROTECTION':
+                node.setmocktime(ACTIVATION_TIME)
+                node.generatetoaddress(11, burn_address)
+                continue
+            tx = CTransaction()
+            num_inputs = len(test_case['sig_hash_types'])
+            spent_outputs = spendable_outputs[:num_inputs]
+            del spendable_outputs[:num_inputs]
+            assert len(spent_outputs) == num_inputs
+            total_input_amount = sum(output.nValue for output in spent_outputs)
+            max_output_amount = (total_input_amount - fee) // test_case['outputs']
+            for i in range(test_case['outputs']):
+                output_amount = max_output_amount - i * 77
+                output_script = CScript([OP_HASH160, i.to_bytes(20, 'big'), OP_EQUAL])
+                tx.vout.append(CTxOut(output_amount, output_script))
+            for _ in test_case['sig_hash_types']:
+                tx.vin.append(CTxIn(COutPoint(int(tx_fan_out.hash, 16), utxo_idx), CScript()))
+                utxo_idx += 1
+            unsigned_tx = tx.serialize().hex()
+            private_keys_wif = []
+            sign_inputs = []
+            for i, spent_output in enumerate(spent_outputs):
+                sign_inputs.append({
+                    'txid': tx_fan_out.hash,
+                    'vout': key_idx + i,
+                    'amount': Decimal(spent_output.nValue) / COIN,
+                    'scriptPubKey': spent_output.scriptPubKey.hex(),
+                })
+            for i, sig_hash_type in enumerate(test_case['sig_hash_types']):
+                sighash = SignatureHashBIP341(
+                    tx_to=tx,
+                    spent_utxos=spent_outputs,
+                    sig_hash_type=sig_hash_type,
+                    input_index=i,
+                    executed_script_hash=hashlib.sha256(executed_scripts[key_idx]).digest(),
+                    codeseparator_pos=test_case.get('codesep', 0xffff_ffff),
+                )
+                if test_case.get('schnorr', False):
+                    signature = private_keys[key_idx].sign_schnorr(sighash)
+                else:
+                    signature = private_keys[key_idx].sign_ecdsa(sighash)
+                signature += bytes([test_case.get('suffix', sig_hash_type & 0xff)])
+                if test_case.get('opcodes'):
+                    tx.vin[i].scriptSig = CScript([signature, executed_scripts[key_idx]])
+                elif test_case.get('is_p2pk'):
+                    tx.vin[i].scriptSig = CScript([signature])
+                else:
+                    tx.vin[i].scriptSig = CScript([signature, public_keys[key_idx].get_bytes()])
```

Comment:
So you add a signature generated in python here.

Reply:
I'm broadcasting it in Line 380/382